### ISSUE TITLE
Add branches to the flat trees

### ISSUE THREAD

**Issue by rmanzoni** [cms-tau-pog/TauTriggerValidation#1]
Hi,

Nice job!

it would be useful to also save:
- decay mode
- bunch crossing (see https://github.com/rmanzoni/cmgtools-lite/blob/80X/H2TauTau/python/proto/analyzers/TreeVariables.py#L19)
- number of primary vertices
- pt of the possibly matched L1 Tau / IsoTau candidate (to also control L1 efficiency at the same time)

Then it may be worth relaxing a some selection criteria and have the possibility of applying them at the plotting stage. For example:
- sign of the mu+tau pair
- transverse mass (we may want to tighten or invert depending on the quantity we want to investigate)
- mass of mu+tau pair

Thanks!
Ric

**Comment by mbluj** [comment]
Hi guys, it is really nice progress, thanks!
What about presenting the tool (+some results) on a coming tau meeting, e.g. on next or next-to-next Monday?

**Comment by l-cadamuro** [comment]
Hi, thanks a lot!
For Riccardo's last points, they are now required in the TauTagAndProbe plugin, so one can simply produce a new collection with these values and remove the "return false" statement.
Same for the additional branches (I think L1 is stored into a separate collection of l1t::Tau, so it requires a separate matching loop but It will be fast to do ;-) )
Michal, would it be fine to present next-to-next Monday? Next week there will be the preapprovals so there's some risk of overlaps.
Cheers,
Luca

**Comment by mbluj** [comment]
Concerning selections at TauTagAndProbe plugin: I agree with Riccardo that it is good to have them reasonably loose to have possibility to check efficiency for different selections (signal-like, W/QCD background enriched, etc.) w/o need of rerunning which takes some time at big datasets.  Another solution can be run a few plugins with different selections in parallel in one job. Anyway, it makes sense to have some basic cuts like mass window, OS on/off, Mt </> a threshold configurable via python cfi. What do you think?

Concerning presentation: both dates are fine, just ask before a weekend to be added to agenda on following Monday.

**Comment by l-cadamuro** [comment]
Hi Michal,

ok agreed, I think each value on which we want to cut can be stored in the output ntuple, and the cut applied at plotting level. This also guarantees that we select only 1 mu-tau pair per event, and only later select candidates from Z->tautau or background.
I will do these changes as soon as possible.

@rmanzoni , can you link/upload the plotting code to see how it can be configured to handle this?

For the presentation, we will ask with the necessary advance!

**Comment by mbluj** [comment]
Yes, right. The most difficult is to make a book keeping which will allow to select only one pair per an event. You can solve this issue either by preforming an arbitration at CMSSW plugin level (as currently done) but it requires some arbitrary decision at early stage or store a couple of pairs (hopefully the number is not high for tight muon not-so-loose tau) and then decide at plotting level.  Finally, you can accept more than one pair per an event for the measurement but it is not preferred solution as it decreases purity.

**Comment by rmanzoni** [comment]
@l-cadamuro have a look here
https://github.com/rmanzoni/triggertools
as is, it is standalone, but it would be nice to have it integrated in this repository itself.
I'll push it under this repository's python folder and let you adjust the imports accordingly.

**Comment by rmanzoni** [comment]
there is some clutter, but that's the best I can do so sneakily 
https://github.com/cms-tau-pog/TauTriggerValidation/tree/master/TauTriggerValidation/python

**Comment by uhussain** [comment]
Hi all,

In the meantime, i’m trying to add the variables suggested by Riccardo to the analyzer and then updating the filters next appropriate to 2016 data. I am figuring this out as I work on this project so i appreciate the input and suggestions.

Thanks,
Usama

**Comment by uhussain** [comment]
Hi everyone, 

I pushed some changes to the repository mainly adding the branches that riccardo mentioned and storing pT, eta, phi for l1taus as well. I made a new branch "mydev" for these changes just in case we need to refine the selection criteria for the matching or need to alter things that i might have added and are not necessary.
